lighthouse-core/report/html/renderer/details-renderer.js

```diff
@@ -55,31 +59,34 @@ class DetailsRenderer {
         // @ts-ignore - TODO(bckenny): Fix type hierarchy
         return this._renderTable(/** @type {TableDetailsJSON} */ (details));
       case 'code':
-        return this._renderCode(details);
+        return this._renderCode(/** @type {StringDetailsJSON} */ (details));
       case 'node':
         return this.renderNode(/** @type {NodeDetailsJSON} */(details));
       case 'criticalrequestchain':
         return CriticalRequestChainRenderer.render(this._dom, this._templateContext,
           // @ts-ignore - TODO(bckenny): Fix type hierarchy
           /** @type {CRCDetailsJSON} */ (details));
+      case 'opportunity':
+        // @ts-ignore - TODO(bckenny): Fix type hierarchy
+        return this._renderTable(/** @type {OpportunityDetails} */ (details));
       default: {
         throw new Error(`Unknown type: ${details.type}`);
       }
     }
   }
 
   /**
-   * @param {NumericUnitDetailsJSON} details
+   * @param {{value: number, granularity?: number}} details
```

Comment:
bytes defaults to granularity of 0.1 in util, so removing the granularity in the audits means these will now show as more precise, right?

dont' we want to keep the granularity the same?

Reply:
> bytes defaults to granularity of 0.1 in util, so removing the granularity in the audits means these will now show as more precise, right? dont' we want to keep the granularity the same?

I forget what we decided on granularity for these. Always to `1` or was bytes going to be one of the auto-formatting options?

Reply:
happy with defaulting opportunity bytes granularity to 1.